browser/ui/views/view_shadow.h

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_VIEW_SHADOW_H_
+#define BRAVE_BROWSER_UI_VIEWS_VIEW_SHADOW_H_
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ref.h"
+#include "ui/compositor/layer_delegate.h"
+#include "ui/compositor/layer_owner.h"
+#include "ui/gfx/geometry/insets.h"
+#include "ui/gfx/shadow_value.h"
+#include "ui/views/view_observer.h"
+
+// Manages a layer that will render a drop shadow under a given view. When an
+// instance is constructed for a view, the view will be set to paint to a layer
+// (if not already set) and a shadow layer will be added below. To remove the
+// shadow, release the `ViewShadow` instance.
+//
+// Example:
+//
+//   class ViewWithShadow : public views::View {
+//    private:
+//     constexpr int kCornerRadius = 8;
+//
+//     constexpr ViewShadow::ShadowParameters kShadow{
+//         .offset_x = 0,
+//         .offset_y = 1,
+//         .blur_radius = 4,
+//         .shadow_color = SkColorSetA(SK_ColorBLACK, 0.07 * 255)};
+//
+//     ViewShadow shadow_{this, kCornerRadius, kShadow};
+//   };
+//
+class ViewShadow : public ui::LayerDelegate, public views::ViewObserver {
```

Comment:
Question: do we like this name? Is it OK to have it in the global namespace?

Reply:
I think so, but you might want to move this class under //brave/ui/views/ and have this under `name views`. Or simply renaming it to DropShadowViews could reduce the chances of collision.

 I'm totally okay with the current state.